src/Adapter/Carrier/CommandHandler/EditCarrierHandler.php

```diff
@@ -75,20 +75,35 @@ public function handle(EditCarrierCommand $command): CarrierId
         if ($command->getLocalizedDelay()) {
             $carrier->delay = $command->getLocalizedDelay();
         }
+        if ($command->getMaxWidth()) {
+            $carrier->width = $command->getMaxWidth();
+        }
+        if ($command->getMaxHeight()) {
+            $carrier->height = $command->getMaxHeight();
+        }
+        if ($command->getMaxDepth()) {
+            $carrier->depth = $command->getMaxDepth();
+        }
+        if ($command->getMaxWeight()) {
+            $carrier->weight = $command->getMaxWeight();
+        }
 
         $this->carrierValidator->validate($carrier);
 
-        $carrierId = $this->carrierRepository->updateInNewVersion($command->getCarrierId(), $carrier);
-
+        $newCarrier = $this->carrierRepository->updateInNewVersion($command->getCarrierId(), $carrier);
+        if ($command->getGroupAccess()) {
+            $this->carrierValidator->validateGroupsExist($command->getGroupAccess());
+            $this->carrierRepository->setGroupsAccess($newCarrier, $command->getGroupAccess());
+        }
         if ($command->getLogoPathName() !== null) {
-            $this->carrierLogoFileUploader->deleteOldFile($carrierId->getValue());
+            $this->carrierLogoFileUploader->deleteOldFile($newCarrier->id);
 
             if ($command->getLogoPathName() !== '') {
                 $this->carrierValidator->validateLogoUpload($command->getLogoPathName());
-                $this->carrierLogoFileUploader->upload($command->getLogoPathName(), $carrierId->getValue());
+                $this->carrierLogoFileUploader->upload($command->getLogoPathName(), $newCarrier->id);
```

Comment:
Same idea for the logo validation, all the validation processes must be done before any update is performed